connectors/sources/gmail.py

```diff
@@ -184,6 +184,7 @@ def _google_directory_client(self):
         return GoogleDirectoryClient(
             json_credentials=self._service_account_credentials,
             customer_id=self.configuration["customer_id"],
+            subject=self.configuration["subject"],
```

Comment:
Question: why not add the `subject` to the json credentials when you create the `self._service_account_credentials` cache? Is it only required in specific cases?

Reply:
We need to re-update it for every user for the GMail client due to this issue https://issuetracker.google.com/issues/290567932?pli=1. If we would cache `subject` in `self._service_account_credentials` it wouldn't fetch messages for users different than the initial `subject`. That's also the reason why the GMail client cannot be cached until this is fixed.